docs/02-app/01-building-your-application/07-configuring/01-typescript.mdx

```diff
@@ -177,19 +177,21 @@ For [`getStaticProps`](/docs/pages/api-reference/functions/get-static-props), [`
 ```tsx filename="pages/blog/[slug].tsx"
 import { GetStaticProps, GetStaticPaths, GetServerSideProps } from 'next'
 
-export const getStaticProps: GetStaticProps = async (context) => {
+export async function getStaticProps(context) {
   // ...
-}
+} satisfies GetStaticProps
```

Comment:
This doesn't work. `satisfies` doesn't support functions (https://github.com/microsoft/TypeScript/issues/51556)

In fact, not even the code in your tweet works. You have to wrap the arrow function in parenthesis, i.e.
```ts
export const getStaticProps = (async (context) => {
  // ...
}) satisfies GetStaticProps
```

Reply:
Thanks, I hadn't tested this yet 😄 Will update.